source/patterns/@aws-solutions-constructs/aws-lambda-secretsmanager/test/lambda-secretsmanager.test.ts

```diff
@@ -0,0 +1,331 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import { Stack } from "@aws-cdk/core";
+import * as lambda from "@aws-cdk/aws-lambda";
+import { Secret } from '@aws-cdk/aws-secretsmanager';
+import * as ec2 from "@aws-cdk/aws-ec2";
+import { LambdaToSecretsmanager } from '../lib';
+import { SynthUtils } from '@aws-cdk/assert';
+import '@aws-cdk/assert/jest';
+
+// --------------------------------------------------------------
+// Test minimal deployment with new Lambda function
+// --------------------------------------------------------------
+test('Test minimal deployment with new Lambda function', () => {
+  // Stack
+  const stack = new Stack();
+  // Helper declaration
+  new LambdaToSecretsmanager(stack, 'lambda-to-secretsmanager-stack', {
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_10_X,
+      handler: 'index.handler',
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`)
+    }
+  });
+  // Assertion 1
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
+});
+
+// --------------------------------------------------------------
+// Test the getter methods
+// --------------------------------------------------------------
+test('Test the properties', () => {
+  // Stack
+  const stack = new Stack();
+  // Helper declaration
+  const pattern = new LambdaToSecretsmanager(stack, 'lambda-to-secretsmanager-stack', {
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_10_X,
+      handler: 'index.handler',
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`)
+    }
+  });
+  // Assertion 1
+  const func = pattern.lambdaFunction;
+  expect(func).toBeDefined();
+  // Assertion 2
+  const secret = pattern.secret;
+  expect(secret).toBeDefined();
+});
+
+// --------------------------------------------------------------
+// Test deployment w/ existing secret
+// --------------------------------------------------------------
+test('Test deployment w/ existing secret', () => {
+  // Stack
+  const stack = new Stack();
+  // Helper declaration
+  const secret = new Secret(stack, 'secret', {});
+  new LambdaToSecretsmanager(stack, 'lambda-to-secretsmanager-stack', {
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_10_X,
+      handler: 'index.handler',
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`)
+    },
+    existingSecretObj: secret
+  });
+  // Assertion 1
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
```

Comment:
Only use the snapshot on the default deployment - use inspection functions for other tests.